samcli/commands/init/init_flow_helpers.py

```diff
@@ -146,11 +146,19 @@ def _get_templates_with_dependency_manager(templates_options, dependency_manager
     return [t for t in templates_options if t.get("dependencyManager") == dependency_manager]
 
 
-def _get_runtime_from_image(image):
+def _get_runtime_from_image(image: str) -> str:
     """
     Get corresponding runtime from the base-image parameter
+
+    Expecting 'amazon/{runtime}-base'
+    But might also be like 'amazon/{runtime}-provided.al2-base'
     """
-    runtime = image[image.find("/") + 1 : image.find("-")]
+    match = re.fullmatch(r"amazon/([a-z0-9.]*)-?([a-z0-9.]*)-base", image)
+    if match is None:
+        return "UNKNOWN"
```

Comment:
Should we return `None` here and handle this at the part where we are calling this method?

Reply:
Sure I can take a swing at it!

I added the "Unknown" return when looking at an un-related unit test failure, where the previous parse before my change resulted in an empty string, and my initial attempt (which threw in case of a regex match failure) caused the test to fail earlier than expected - so hoping that's not a sign that the proposed change ends up being intrusive 🤞